judgels-backends/sandalphon/sandalphon-api/src/main/java/judgels/sandalphon/api/problem/bundle/Item.java

```diff
@@ -9,7 +12,25 @@
     String getJid();
     ItemType getType();
     String getMeta();
-    String getConfig();
+
+    @JsonTypeInfo(
+            use = JsonTypeInfo.Id.NAME,
+            include = JsonTypeInfo.As.EXTERNAL_PROPERTY,
+            property = "type",
+            visible = true
+    )
+    @JsonSubTypes({
+            @JsonSubTypes.Type(value = ImmutableStatementItemConfig.class, name = ItemType.Statement),
+            @JsonSubTypes.Type(value = ImmutableMultipleChoiceItemConfig.class, name = ItemType.MultipleChoice)
+    })
+    ItemConfig getConfig();
+
+    default Item processDisplayText(Function<String, String> processor) {
```

Comment:
This can be removed, to minimize logic in a POJO. We can do this in the resource class instead. (Like what you do for `withoutGradingInfo()`

Reply:
Replaced with Processor pattern (like in `ScoreboardProcessor`)